crates/shared/src/signature_validator/simulation.rs

```diff
@@ -93,25 +93,46 @@ impl Validator {
             (self.settlement.address(), self.vault_relayer),
             check.signer,
             Bytes(check.hash),
-            Bytes(check.signature),
+            Bytes(check.signature.clone()),
             check
                 .interactions
-                .into_iter()
-                .map(|i| (i.target, i.value, Bytes(i.call_data)))
+                .iter()
+                .map(|i| (i.target, i.value, Bytes(i.call_data.clone())))
                 .collect(),
         );
-        let response_bytes = self
+        let simulation = self
             .settlement
             .simulate_delegatecall(
                 self.signatures.address(),
                 Bytes(validate_call.tx.data.unwrap_or_default().0),
             )
-            .from(crate::SIMULATION_ACCOUNT.clone())
+            .from(crate::SIMULATION_ACCOUNT.clone());
+
+        let result = simulation
+            .clone()
             .call_with_state_overrides(&overrides)
-            .await?;
+            .await;
+
+        let response_bytes = match result {
+            Ok(bytes) => bytes,
+            Err(err) => {
+                tracing::debug!(
+                    ?simulation,
+                    ?check,
+                    ?overrides,
+                    "signature verification failed"
+                );
+                return Err(err.into());
+            }
+        };
```

Comment:
The ? should convert the error too

```suggestion
        let response_bytes = result.inspect_err(|e| tracing::debug!(
                    ?simulation,
                    ?check,
                    ?overrides,
                    "signature verification failed"
                ))?;
```